multicluster/controllers/multicluster/core/remote_cluster.go

```diff
@@ -0,0 +1,353 @@
+	config.CAData = crtData
+	mgr, err := ctrl.NewManager(config, ctrl.Options{
+		Scheme:             scheme,
+		MetricsBindAddress: "0",
+		Namespace:          clusterSetNamespace,
+	})
+	if err != nil {
+		klog.Error(err, "unable to create new manager", "cluster", clusterID)
+		return nil, err
+	}
+
+	remoteClient, e := client.New(config, client.Options{Scheme: scheme})
+	if e != nil {
+		return nil, e
+	}
+	cluster := &remoteCluster{
+		Client:               remoteClient,
+		ClusterManager:       mgr,
+		ClusterSetID:         clusterSetID,
+		ClusterID:            clusterID,
+		config:               config,
+		scheme:               scheme,
+		Namespace:            clusterSetNamespace,
+		connected:            false,
+		localClusterManager:  localClusterManager,
+		remoteClusterManager: remoteClusterManager,
+	}
+
+	remoteClusterManager.AddRemoteCluster(cluster)
+
+	return cluster, nil
+}
+
+/**
+ * When a member is added to a ClusterSet, a specific ServiceAccount is created on the
+ * leader cluster which allows the member to access into the common area. This ServiceAccount
+ * has an associated Secret which must be copied into the Member Cluster as an opaque secret.
+ * Name of this secret is part of the ClusterSet spec for this member. This method reads
+ * the Secret given by that name.
+ */
+func getSecretCACrtAndToken(namespace string, secretName string) ([]byte, []byte, error) {
+	var err error
+	mutex.Lock()
+	if secretClient == nil {
+		secretClient, err = client.New(ctrl.GetConfigOrDie(), client.Options{})
+		if err != nil {
+			mutex.Unlock()
+			return nil, nil, err
+		}
+	}
+	mutex.Unlock()
+	secretObj := &v1.Secret{}
+	secretNamespacedName := types.NamespacedName{
+		Namespace: namespace,
+		Name:      secretName,
+	}
+	err = secretClient.Get(context.TODO(), secretNamespacedName, secretObj)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	caData, found := secretObj.Data[v1.ServiceAccountRootCAKey]
+	if !found {
+		return nil, nil, fmt.Errorf("ca.crt data not found in secret %v", secretName)
+	}
+
+	token, found := secretObj.Data[v1.ServiceAccountTokenKey]
+	if !found {
+		return nil, nil, fmt.Errorf("token not found in secret %v", secretName)
+	}
+
+	return caData, token, nil
+}
+
+func (r *remoteCluster) SendMemberAnnounce() error {
+	memberAnnounceList := &multiclusterv1alpha1.MemberClusterAnnounceList{}
+	if err := r.List(context.TODO(), memberAnnounceList, client.InNamespace(r.GetNamespace())); err != nil {
+		return err
+	}
+	var localClusterMemberAnnounce multiclusterv1alpha1.MemberClusterAnnounce
+	localClusterMemberAnnounceExist := false
+	if len(memberAnnounceList.Items) != 0 {
+		for _, memberAnnounce := range memberAnnounceList.Items {
+			if memberAnnounce.ClusterID == string(r.remoteClusterManager.GetLocalClusterID()) {
+				localClusterMemberAnnounceExist = true
+				localClusterMemberAnnounce = memberAnnounce
+				break
+			}
+		}
+	}
+	ctx, cancelFunc := context.WithTimeout(context.Background(), 5*time.Second)
```

Comment:
@luolanzone @tnqn : do we have a better way to achieve this?

Reply:
Removed it for now